compiler/rustc_const_eval/src/lib.rs

```diff
@@ -59,3 +59,49 @@ pub fn provide(providers: &mut Providers) {
         const_eval::deref_mir_constant(tcx, param_env, value)
     };
 }
+
+use crate::const_eval::CompileTimeInterpreter;
+use crate::interpret::{InterpCx, MemoryKind, OpTy};
+use rustc_middle::ty::layout::LayoutCx;
+use rustc_middle::ty::{layout::TyAndLayout, ParamEnv, TyCtxt};
+use rustc_session::Limit;
+use rustc_span::Span;
+use rustc_target::abi::InitKind;
```

Comment:
Please no imports in the middle of the file.
Also this file shouldn't contain any logic, add a new file for that please.